mobile/src/navigations/main/main.tsx

```diff
@@ -1,30 +1,85 @@
-import {
-  type BottomTabNavigationOptions,
-  createBottomTabNavigator,
-} from '@react-navigation/bottom-tabs';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { getHeaderTitle } from '@react-navigation/elements';
+import { type RouteProp } from '@react-navigation/native';
 import React from 'react';
+import { type StyleProp, type ViewStyle } from 'react-native';
 
+import ChatIcon from '#assets/img/icons/chat.svg';
 import HomeIcon from '#assets/img/icons/home.svg';
 import { AppColor, MainScreenName } from '#libs/enums/enums';
-import { type TabNavigationParameterList } from '#libs/types/types';
-import { Home } from '#screens/main/home';
+import {
+  type TabNavigationParameterList,
+  type TabNavigatorOptions,
+} from '#libs/types/types';
+import { Chat } from '#screens/chat/chat';
+import { ChatHeader } from '#screens/chat/components/components';
+import { HomeHeader } from '#screens/home/components/components';
+import { Home } from '#screens/home/home';
 
 import { styles } from './styles';
 
 const BottomTab = createBottomTabNavigator<TabNavigationParameterList>();
 
-const tabNavigatorOptions: BottomTabNavigationOptions = {
-  tabBarActiveTintColor: AppColor.BLUE_200,
-  tabBarIcon: ({ color }) => <HomeIcon color={color} />,
-  tabBarInactiveTintColor: AppColor.GRAY_300,
-  tabBarShowLabel: false,
-  tabBarStyle: styles.tabBarStyle,
-};
-
 const Main: React.FC = () => {
+  const tabNavigatorOptions = ({
+    route,
+  }: {
+    route: RouteProp<TabNavigationParameterList>;
+  }): TabNavigatorOptions => ({
+    tabBarActiveTintColor: AppColor.BLUE_300,
+    tabBarIcon: ({ color }: { color: string }): React.ReactNode => {
+      switch (route.name) {
+        case MainScreenName.HOME: {
+          return <HomeIcon color={color} />;
+        }
+        case MainScreenName.CHAT: {
+          return <ChatIcon color={color} />;
+        }
+      }
+    },
+    headerStyle: styles.headerStyle,
+    headerTitleStyle: styles.headerTitleStyle,
+    tabBarInactiveTintColor: AppColor.GRAY_300,
+    tabBarShowLabel: false,
+    tabBarStyle: styles.tabBarStyle,
+  });
+
   return (
     <BottomTab.Navigator screenOptions={tabNavigatorOptions}>
-      <BottomTab.Screen name={MainScreenName.HOME} component={Home} />
+      <BottomTab.Screen
+        name={MainScreenName.HOME}
+        component={Home}
+        options={{
+          header: ({ route, options }): React.ReactNode => {
+            const title = getHeaderTitle(options, route.name);
+
+            return (
+              <HomeHeader
+                title={title}
+                headerStyle={options.headerStyle as StyleProp<ViewStyle>}
+                titleStyle={options.headerTitleStyle as StyleProp<ViewStyle>}
+              />
+            );
+          },
+        }}
```

Comment:
Can we use only 1 Header component? I think we can have the similar headers for Home and Chat screens.

Reply:
Done

Reply:
Maybe we don't even need to create a HeaderTitle component at all. For this navigation file, we can use headerStyle in screenOptions (I don't think we will have lots of styles), the header title will be the screen name (which is by default) and for the badge in Chat component we can use ```navigation.setOptions({...})``` inside of Chat component.
This way the components will be cleaner and we will not have to make a request to the backend to get the number of chats when we are in other screens (for example in home screen).

@Liza-Veis @v0ldemar01 what do you think?

Reply:
I totally agree with Farid

Reply:
agree

Reply:
@fshabanov could you please clarify one question for me?

I have removed HeaderTitle from the main.tsx and added Badge to the chat.tsx.
But I did it like this:

chat.tsx:
`navigation.setOptions({
      headerTitle: ({ children }: { children: React.ReactNode }) => (
        <View style={styles.headerTitleWrapper}>
          <Text style={styles.headerTitle}>{children}</Text>
          <Badge />
        </View>
      ),
});`

I have tried to use the headerRight/headerLeft property, but I havn't figured out how to set the position of the badge to be close to the title. So for now I still set Title manualy (using children) but not in main.tsx. Is it ok or should I change it?

Reply:
For me it's ok

Reply:
I agree, I think it's okay